splink/internals/expectation_maximisation.py

```diff
@@ -169,7 +169,7 @@ def populate_m_u_from_lookup(
 ) -> None:
     cl = comparison_level
 
-    if "m" not in training_fixed_probabilities:
+    if not cl._fix_m_probability and "m" not in training_fixed_probabilities:
```

Comment:
This is needed because otherwise the cl.m_probability in the training linker is set, and so the value fluctuates during EM training despite not being assigned back to the m probability on the main linker